Storage/tests/System/ManageObjectsTest.php

```diff
@@ -414,4 +444,22 @@ public function testStringNormalization()
             $this->assertSame($expectedContent, $actualContent);
         }
     }
+
+    private function assertStorageObjectExists($object, $options = [], $isPresent = true)
+    {
+        // $this->assertEquals($isPresent, $object->exists($options));
```

Comment:
I am observing that restored object fails the exist check while a new object instantiated from bucket passes the exists check.

Reply:
Fixed it in 162afbd159b17d0df708acae79b0d44a46e3cb45